### PR TITLE
Set alt-text for the feed-image if no other text is displayed (a11y)

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue # .

### Summary of Changes
As title says

### Testing Instructions
Make a module syndication feeds. 
Try different variants: 

- Display text yes, and enter a text
- Display text yes, but let the text empty
- Display text no

Inspect the code for the image and see which alt text is displayed.

### Actual result BEFORE applying this Pull Request
The image always has a hard coded alt-text. 

### Expected result AFTER applying this Pull Request
If the feed image is displayed with a text, then it has alt="", 
otherwise it has a translated text "feed entries"

### Documentation Changes Required
no

### CODE REVIEW

File: modules/mod_syndicate/tmpl/default.php [hunk 1]
```diff
@@ -3,15 +3,17 @@
  * @package     Joomla.Site
  * @subpackage  mod_syndicate
  *
- * @copyright   Copyright (C) 2005 - 2020 Open Source Matters, Inc. All rights reserved.
+ * @copyright   (C) 2006 Open Source Matters, Inc. <https://www.joomla.org>
```
Comment (hans2103) [inline modules/mod_syndicate/tmpl/default.php]: I don't think this change is needed

File: modules/mod_syndicate/tmpl/default.php [hunk 2]
```diff
@@ -3,15 +3,17 @@
...
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
 
 defined('_JEXEC') or die;
 
+// Set alt-text for the image only if no text is displayed
+$alt = (!$params->get('display_text', 1) && empty($text)) ? JText::_('MOD_SYNDICATE_DEFAULT_FEED_ENTRIES') : '';
```
Comment (hans2103) [inline modules/mod_syndicate/tmpl/default.php]: Please replace `JText` by `Text`and call for `use Joomla\CMS\Language\Text;`

File: modules/mod_syndicate/tmpl/default.php [hunk 3]
```diff
@@ -3,15 +3,17 @@
...
 ?>
 <a href="<?php echo $link; ?>" class="syndicate-module<?php echo $moduleclass_sfx; ?>">
-	<?php echo JHtml::_('image', 'system/livemarks.png', 'feed-image', null, true); ?>
+	<?php echo JHtml::_('image', 'system/livemarks.png', $alt, null, true); ?>
```
Comment (hans2103) [inline modules/mod_syndicate/tmpl/default.php:16]: Please replace `JHtml` by `HTMLHelper` and call for use `Joomla\CMS\HTML\HTMLHelper;`

**Review by hans2103 — CHANGES_REQUESTED** [review]

File: modules/mod_syndicate/tmpl/default.php (hunk 2 above)
Comment (chmst) [reply modules/mod_syndicate/tmpl/default.php]: thank you, fixed

File: modules/mod_syndicate/tmpl/default.php (hunk 3 above)
Comment (chmst) [reply modules/mod_syndicate/tmpl/default.php:16]: This is staging and must not be changed.

**Review by hans2103 — APPROVED** [review]
namespacing is J4. My bad. Sorry

**Comment by hans2103** [comment]
I have tested this item :white_check_mark: successfully on dfcd22fd28bd59c8a1106b2f672148bb49208a15<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31180">issues.joomla.org/tracker/joomla-cms/31180</a>.</sub>

**Comment by adj9** [comment]
I have tested this item :white_check_mark: successfully on dfcd22fd28bd59c8a1106b2f672148bb49208a15<br /><br />done
Nominal pose modul: top-a

HTML inspect:

- with lable = n 
Schermata 2020-10-21 alle 12 52 56


- with confirmed lable but = ""
Schermata 2020-10-21 alle 12 55 17

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31180">issues.joomla.org/tracker/joomla-cms/31180</a>.</sub>

**Comment by chmst** [comment]
@adj9 you have made yor test on J4 :) 

It works on both versions, but the JText_:: (in 3.9) and TEXT::_  are different<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31180">issues.joomla.org/tracker/joomla-cms/31180</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on dfcd22fd28bd59c8a1106b2f672148bb49208a15<br /><br />The test for empty text is not correct. 

To replicate the error first create a module with display text set to yes and a value in feed entries
Save and check it - all good

Now change the module to display text no - save and check
Now we do not get an alt text because there is a value for $text saved in the module<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31180">issues.joomla.org/tracker/joomla-cms/31180</a>.</sub>

**Comment by ghost** [comment]
@chmst ?

**Comment by chmst** [comment]
Thank you for remembering me. Fixed now. Could you please re-test?

**Comment by chmst** [comment]
Fixed in J4, no need to change something in J3.

**Pull request closed without merging**